src/module.cpp

```diff
@@ -476,7 +476,6 @@ instruction_ref module::insert_literal(instruction_ref ins, literal l)
 
 instruction_ref module::insert_parameter(instruction_ref ins, std::string name, shape s)
 {
-    assert(get_parameter_shape(name) == shape{});
```

Comment:
We shouldn't remove this assert. We should probably rename the parameter before we insert the new one.